WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostListViewModel.kt

```diff
@@ -197,7 +200,24 @@ class ReaderPostListViewModel @Inject constructor(
         updateSubfilter(filter)
     }
 
-    fun changeSubfiltersVisibility(show: Boolean) = _shouldShowSubFilters.postValue(show)
+    fun changeSubfiltersVisibility(show: Boolean) {
+        if (show) {
+            if (getCurrentSubfilterValue().isTrackedItem &&
+                        !readerTracker.isRunning(ReaderTrackerType.SUBFILTERED_LIST)) {
+                AppLog.d(T.READER, "TRACK READER ReaderPostListFragment > START Count SUBFILTERED_LIST")
+                readerTracker.start(ReaderTrackerType.SUBFILTERED_LIST)
+            } else if (!getCurrentSubfilterValue().isTrackedItem &&
+                    readerTracker.isRunning(ReaderTrackerType.SUBFILTERED_LIST)) {
+                AppLog.d(T.READER, "TRACK READER ReaderPostListFragment > STOP Count SUBFILTERED_LIST")
+                readerTracker.stop(ReaderTrackerType.SUBFILTERED_LIST)
+            }
+        } else if (readerTracker.isRunning(ReaderTrackerType.SUBFILTERED_LIST)) {
+            AppLog.d(T.READER, "TRACK READER ReaderPostListFragment > STOP Count SUBFILTERED_LIST")
+            readerTracker.stop(ReaderTrackerType.SUBFILTERED_LIST)
+        }
```

Comment:
This is kind of hard to read at first glance. What do you think about extracting a couple of local variables to the top of the function to make it a bit easier to parse? 

I'm thinking something like this:

```kotlin
val isCurrentItemTracked = getCurrentSubfilterValue().isTrackedItem
val isTrackerReaderRunning = readerTracker.isRunning(ReaderTrackerType.SUBFILTERED_LIST)
if (show) {
    if (isCurrentItemTracked && !isTrackerReaderRunning) {
        // ...
    } else if (!isCurrentItemTracked && isTrackerReaderRunning) {
        // ...
    }
} else if (isCurrentItemTracked) {
        // ...
}
```

Reply:
Agree 😊! Changed it [here](56f01bceb31845b0c9a545c26dedf0a39fd75fc7)